tool_helper_utils/__init__.py

```diff
@@ -2,12 +2,30 @@
 import sys
 
 
+class AbstractBaseFactoryFactory:
+    def build():
+        return None
+
+
+class FactoryFactory(AbstractBaseFactoryFactory):
+    def __init__(self, factory):
+        self.factory = factory
+
+    def build(self):
+        return self.factory.get_bad_idea()
+       
+
+class FactoryBadIdea:
+    def get_bad_idea(self):
+        return "factory"
+
+
 class StopItImportHook(importlib.abc.MetaPathFinder, importlib.abc.Loader):
     BAD_IDEAS = (
         "tool",
         "helper",
         "util",
-        "factory",
+        FactoryFactory(FactoryBadIdea()).build(),
```

Comment:
We may want to defer the specifics of instantiation to the runtime environment. Reading the name of the particular class to be created from an XML file could work as a stop-gap solution, but in the long run I'd really prefer to see a proper dependency injection mechanism. In particular, the ability to dynamically download additional `AbstractBaseFactoryFactory` subclasses from the Internet could come in handy in ways we can't yet anticipate.